tests/test_models.py

```diff
@@ -396,3 +430,10 @@ def check_insights_data(self, insights, uuid):
         self.assertIsInstance(
             insights.warnings, tuple, "warnings is a tuple, not a dict"
         )
+
+    def check_risk_score_reasons_data(self, reasons):
+        self.assertEqual(45, reasons[0].multiplier)
+        self.assertEqual("ANONYMOUS_IP", reasons[0].reasons[0].code)
```

Comment:
We could add length checks anywhere we are also checking e.g. `reasons[0]` or `reasons[0].reasons[0]`